### PR TITLE
Add `InjectEnumSet` and `InjectSchemaTree` to `LangMapperBase`.

### PR DESCRIPTION
**Pull request by wenovus**
This allows unit tests to be created by custom LangMappers on their methods (e.g. `FieldName`, `KeyLeafType`).

### CODE REVIEW

File: ygen/ir.go [hunk 1]
```diff
@@ -98,12 +98,24 @@ type LangMapperBaseSetup interface {
...
+	setEnumSet(*enumSet)
 
-	// SetSchemaTree is used to supply a copy of the YANG schema tree to
+	// setSchemaTree is used to supply a copy of the YANG schema tree to
 	// the mapped such that leaves of type leafref can be resolved to
 	// their target leaves.
-	SetSchemaTree(*schemaTree)
+	setSchemaTree(*schemaTree)
+
+	// SetupEnumSet is intended to be called by unit tests in order to set up the
```
Comment (robshakir) [inline ygen/ir.go:108]: Should we call this `InjectEnumSet` or something similar to show that it is being used specifically for that purpose? `Setup...` seems to indicate to me that this is something that would be required for non-testing purposes.

File: ygen/ir.go [hunk 2]
```diff
@@ -119,36 +131,66 @@ type LangMapperBase struct {
 	schematree *schemaTree
 }
 
-// SetEnumSet is used to supply a set of enumerated values to the
+// setEnumSet is used to supply a set of enumerated values to the
 // mapper such that leaves that have enumerated types can be looked up.
 //
 // NB: This method is a set-up method that the user does not need to be invoked
```
Comment (robshakir) [inline ygen/ir.go:137]: Just reading this back `a set-up method that the user _does not_ need to be invoked`? Does this mean that the user doesn't need to invoke it, or that they don't care whether this is called? It's a bit confusing.

**Review by robshakir — CHANGES_REQUESTED** [review]

File: ygen/ir.go (hunk 1 above)
Comment (wenovus) [reply ygen/ir.go]: Done.

File: ygen/ir.go (hunk 2 above)
Comment (wenovus) [reply ygen/ir.go]: Fixed up comment, PTAL if this makes sense now.

**Pull request merged**